source/simulator/core.cc

```diff
@@ -1622,6 +1738,55 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Advection_only:
+        {
+          // Identical to IMRES except does not solve Stokes equation
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
```

Comment:
I am a bit afraid of what happens to the free surface with a prescribed stokes solution, did you run any tests? But I guess, it is in the responsibility of the user to select a stokes solution that does not completely twist the surface.